src/c++/perf_analyzer/docs/examples/profile.py

```diff
@@ -0,0 +1,109 @@
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import json
+import random
+import subprocess
+from pathlib import Path
+
+RANDOM_WORDS = [
+    "system",
+    "plug",
+    "gentle",
+    "efficient",
+    "library",
+    "tested",
+    "careful",
+    "sneeze",
+    "excuse",
+    "zoo",
+    "rock",
+    "delight",
+    "hammer",
+    "unit",
+    "happen",
+    "multiply",
+    "texture",
+    "tired",
+    "knot",
+    "yawn",
+]
+
+
+def profile(args):
+    command = (
+        f"perf_analyzer -m {args.model} -i grpc --async --streaming "
+        "--input-data=prompts.json "
+        "--profile-export-file=profile_export.json "
+        "--measurement-mode=count_windows "
+        "--measurement-request-count=10 "
+        "--stability-percentage=999"
+    )
+    ret = subprocess.run(args=[command], shell=True)
+    if ret.returncode == 0:
+        # Example json demonstrating format:
+        #   see client/src/c++/perf_analyzer/docs/examples/decoupled_output_file.json
+        with open("profile_export.json") as f:
+            requests = json.load(f)["experiments"][0]["requests"]
+            latencies = [r["response_timestamps"][0] - r["timestamp"] for r in requests]
+            avg_latency_s = sum(latencies) / len(latencies) / 1_000_000_000
+        return avg_latency_s
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-m",
+        "--model",
+        type=str,
+        default="vllm",
+        help="The name of the model to profile.",
+    )
+    args = parser.parse_args()
+
+    prompt_lengths = [10, 100, 500, 800, 1000]
+    input_data = {"data": [{"STREAM": [True]}]}
+    results = []
+
+    for prompt_length in prompt_lengths:
```

Comment:
It seems like this is generating a prompt of L words (each of which has a random length). Two things:

1. I thought the prompt length was the absolute length of the prompt
2. Since the words are random in length and chosen randomly, doesn't that mean from run to run the prompt will be of different lengths?

Reply:
Ah, I thought prompt length meant number of characters in the prompt. Is this something that Triton/PA has defined or is this a well known term in LLM usage?

Are different words required or could we just repeat the same word over and over again? My worry is the possibility of getting variations between the runs of the same prompt length if we didn't choose the same sequence of words.

Reply:
It's not something Triton/PA defined. I think the more precise term is "token", which is not quite the same as word, but both are just used interchangeably.

I think you're raising a great point. Since a single word may not always be a single token, I think it would be better to use a single common word (e.g. "hi") to ensure the number of words == number of tokens.

Reply:
Changed to use a single word to generate prompt